### DISCUSSION TITLE
Support for OAuth Client Credentials Flow

### DISCUSSION THREAD

**Discussion by bakejeyner** [nextauthjs/next-auth#804] [Help]
**Your question**
I was wondering if there was existing support for the [OAuth Client Credentials Flow](https://tools.ietf.org/html/rfc6749#section-4.4). More specifically, to have the nextjs app act as `Your API` in [this OAuth Client Credentials Flow Diagram](https://auth0.com/docs/flows/client-credentials-flow#how-it-works).

**What are you trying to do**
Every request coming into my nextjs app will have an access token found in the `Authorization` header of that request. I want next-auth to pull that access token off of the incoming request, verify the access token against its configured providers, and if valid convert the access token to a next-auth session for use throughout the nextjs app.

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

I spent a couple of hours researching this use case but was unable to find directly relevant information. My best idea was to create a provider who asked for the `client_credentials` grant type, but even with an access token present on the incoming request next-auth would always ask me to sign into a provider.

* [ ] Found the documentation helpful
* [x] Found documentation but was incomplete
* [ ] Could not find relevant documentation
* [ ] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by danielweil** [comment]
Have you found a solution?

**Reply by bakejeyner** [reply]
> Have you found a solution?

I have not. We still require that the user sign into the configured provider directly instead of using inbound credentials.
